yt_dlp/extractor/screencastomatic.py

```diff
@@ -32,8 +34,29 @@ class ScreencastOMaticIE(InfoExtractor):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        password = self.get_param('videopassword')
         webpage = self._download_webpage(
             'https://screencast-o-matic.com/player/' + video_id, video_id)
+
+        if (self._html_extract_title(webpage) == 'Protected Content'
+                or 'This video is private and requires a password' in webpage):
+            if password:
+                form = self._search_regex(
+                    r'(?is)<form[^>]*>(?P<form>.+?)</form>', webpage, '%s form', group='form')
```

Comment:
Avoid `.+?`. This can cause catastrophic backtracking, and also breaks if the page has multiple forms

Reply:
what's `%s form`?

Reply:
I was adapting the `_form_hidden_inputs` function in `common.py`. I've replaced `%s form`. Do you have any suggestions for replacing `.+?`? It seems to work when there are multiple forms because of the non-greedy modifier, but I see your point about catastrophic backtracking.

Reply:
Thinking about it more, I see how this is pretty much the only way

Reply:
btw, does the form not have an id? If it does, why not use `_form_hidden_inputs`?

Reply:
Yeah, the form doesn't have an id, hence the hack. I guess we could also try generalizing `_form_hidden_inputs`?

Reply:
This pattern doesnt seem to appear often. So not worth it imo